src/interfaces/chain.h

```diff
@@ -269,6 +269,9 @@ class Chain
     //! Register handler for RPC. Command is not copied, so reference
     //! needs to remain valid until Handler is disconnected.
     virtual std::unique_ptr<Handler> handleRpc(const CRPCCommand& command) = 0;
+
+    //! Notify handler of all memory pool transactions.
+    virtual void notifyMempoolTransactions(Notifications& notifications) = 0;
```

Comment:
nit: I'd prefer the name `getMempoolTransactions()` or `fetchMempoolTransactions()`. Notify to me signifies that the wallet is subscribing and getting a notification, whereas this interface function is actually used by the wallet to actively fetch the mempool transactions.

Reply:
> I'd prefer the name `getMempoolTransactions()` or `fetchMempoolTransactions()`.

I think these are confusing since the function is void.

Reply:
> Notify to me signifies that the wallet is subscribing and getting a notification

The "standard" for that is `handleX`, like `handleNotifications`.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15652#discussion_r269275899

Agree with John the name is not great. I think I'd probably call it `pushMempoolNotifications`.

More importantly, though, I think this needs to have a comment saying how it relates to `handleNotifications`. A good comment might be:

"Synchronously send TransactionAddedToMempool notifications about all current mempool entries to the specified handler. To avoid race conditions where out of date TransactionAddedFromMempool / TransactionRemovedFromMempool asynchronous notifications previously requested through `handleNotifications` would arrive during or after this call, this will block waiting for any previous asynchronous notifications to complete, and prevent new asynchronous notifications from being sent until this returns."

I will say writing this makes me think the interface is overcomplicated. I guess I'd prefer my original suggestion of just sending `TransactionAddedFromMempool` notifications atomically in the same place the handler is registered: https://github.com/bitcoin/bitcoin/pull/15652#discussion_r268675190. Marco pointed out this would be wasteful in the case of `createwallet` calls https://github.com/bitcoin/bitcoin/pull/15652#discussion_r268710750, but the notifications could be optional if the waste is too much.

Reply:
`requestMempoolTransactions()`?

Agree with Russ about adding a comment.

Reply:
> just sending TransactionAddedFromMempool notifications atomically in the same place the handler is registered

I understand it is cleaner but it is weird to have receive notifications before `handleNotifications` return. Thanks for the comment!

Eeny, meeny, miny, moe `requestMempoolTransactions`, after all we want to be notified of mempool transactions.

Reply:
> but it is weird to receive notifications before handleNotifications return

This is the current behavior and it isn't weird. Notifications start to arrive when RegisterValidationInterface is called, and can happen before that function returns, much less before handleNotifications returns.

Reply:
> This is the current behavior

Currently `cs_main` and cs_wallet` are locked when `handleNotifications` is called so that doesn't happen, but otherwise you are right.